torch/_inductor/codegen/aoti_runtime/interface.cpp

```diff
@@ -26,6 +26,17 @@
             std::to_string(actual_size));                         \
   } while (0)
 
+// AOTInductor uses at::addmm_out, which doesn't supports
+// arguments that requires gradient. For this reason, we
+// enforce no_grad context for run APIs.
+#define WITH_NO_GRAD(...)                               \
+  do {                                                  \
+    bool prev_mode = aoti_torch_grad_mode_is_enabled(); \
+    aoti_torch_grad_mode_set_enabled(false);            \
+    __VA_ARGS__                                         \
+    aoti_torch_grad_mode_set_enabled(prev_mode);        \
```

Comment:
Wondering if we need some sort of RAII to restore the previous grad mode in case an exception occurs?